LibertyDSNP/example-client
# Add GraphChangeButton to Profile

Purpose
---------------
We would like to be able to follow users whose content is posted in our general feed. We currently only have follow buttons in the connections list section, which means that we can only follow users who already follow us (and unfollow those users or the users we already follow). Now that we can view user's profiles by clicking on the header of a post, we can place a graph change button in the profile block to follow these users.


Solution
---------------
Add a GraphChangeButton to the profile page if it is not displaying the current user.

Change summary
---------------
* Add GraphChangeButton to the profile page.
* Fix the connection list so it displays the follows and followed by's of the other user (this was missed in earlier PRs)
* Refactor the styling so GraphChangeButton gets its own styles.


Steps to Verify
----------------
1. Start with a clean,  populated app.
2. In the general feed, scroll down the the first post by a user that is not the current user.
3. Click the header of the post to view the profile of this user on the right.
4. Note the new follow/unfollow button under the icon of the selected user in profile block to the right.
5. Keep clicking on post headers till you find a user that you are not currently following.
5. Click the button to follow this user, accept the signature requests, and wait for the transaction to complete.
6. Note that the button changes state.
7. Click the "back" button in the feed tab to show the current user profile.
8. Note that the followed user appears in the "following" tab and their posts appear in your feed.
9. Return to the new followed user's profile.
10. Click "unfollow" and complete transaction.
11. Confirm user is unfollowed in all the same places.


Additional details / screenshot
----------------

-<img width="472" alt="Screen Shot 2021-09-07 at 12 36 38 PM" src="https://user-images.githubusercontent.com/169613/132380488-b00fd963-2481-475b-98bd-7a43c21d024c.png">
